NSS-Day-Cohort-69/tabloid-one-more-div
# Feature - Posts List View

### Purpose
To allow all logged in Users to view a list of all approved and published posts

### Files Changed
- Reorganized the following files: `CategoryDTO.cs`, `CategoryNoNavDTO.cs`, `CommentDTO.cs`, `PostDTO.cs`, `PostReactionDTO.cs`, `PostTagDTO.cs`, `ReactionDTO.cs`, `ReactionNoNavDTO.cs`, `SubscriptionDTO.cs`, `TagDTO.cs`, `TagNoNavDTO.cs`, `UserProfileDTO.cs`, and `TagList.jsx`
- Created initial User Profile DTO used for Comments in new `UserProfileForCommentDTO.cs` DTO class
- Created initial User Profile DTO used for Posts in new `UserProfileForPostDTO.cs` DTO class
- Created initial User Profile DTO used for Post Reactions in new `UserProfileForPostReactionDTO.cs` DTO class
- Created initial Post Reaction DTO used for Posts in new `PostReactionForPostDTO.cs` DTO class
- Created initial Comment DTO used for Posts in new `CommentForPostDTO.cs` DTO class
- Updated properties to use custom DTOs in `PostDTO.cs`
- Created initial Post DTO used when creating a list of Posts that don't contain their content in new `PostForListDTO.cs`
- Created file and wrote endpoint that returns a list of all approved and published Posts in new `PostController.cs` controller
- Created file and wrote fetch for getting all approved and published Posts in new `postManager.js`
- Created file and wrote reusable page container element in new `PageContainer.jsx` component module
- Created file and wrote initial logic and rendering for Posts List view in new `PostList.jsx` component module
- Added routing for Posts List view in `ApplicationViews.jsx`
- Added nav button for Posts List view in `NavBar.jsx`

### To Test
Fetch, setup, and run according to project README, but do not create a new migration, and confirm the following
- Logged in Users have a "Posts" button in the navbar that navigates the user to the Posts List view
- Posts List view shows all approved and published Posts in the order of newest to oldest

closes #44
